src/download.cpp

```diff
@@ -24,29 +24,58 @@ Download::Download(std::function<void()> cb_require_view_update_)
 
 Download::~Download() {}
 
-const std::string Download::filename() const
+Download::Download(const Download& download) :
+	url_(download.url_),
+	download_status(download.download_status),
+	msg(download.msg),
+	cursize(download.cursize),
+	totalsize(download.totalsize),
+	curkbps(download.curkbps),
+	offs(download.offs),
+	cb_require_view_update(download.cb_require_view_update)
 {
-	return fn;
+	fn = download.fn.clone();
 }
 
-const std::string Download::basename() const
+Download& Download::operator=(const Download& download)
 {
-	std::string::size_type start = fn.rfind(NEWSBEUTER_PATH_SEP);
+	if (this == &download) {
+		return *this;
+	}
+	url_ = download.url_;
+	download_status = download.download_status;
+	msg = download.msg;
+	cursize = download.cursize;
+	totalsize = download.totalsize;
+	curkbps = download.curkbps;
+	offs = download.offs;
+	fn = download.fn.clone();
+	return *this;
+}
+
+const newsboat::Filepath Download::filename() const
+{
+	return fn.clone();
+}
+
+const newsboat::Filepath Download::basename() const
+{
+	std::string::size_type start = fn.to_locale_string().rfind(NEWSBEUTER_PATH_SEP);
 
 	if (start != std::string::npos) {
-		return fn.substr(start+1);
+		return fn.to_locale_string().substr(start+1);
 	}
-	return fn;
+	return fn.clone();
```

Comment:
Please add a `file_name` method to `Filepath` that calls this Rust method: https://doc.rust-lang.org/nightly/std/path/struct.Path.html#method.file_name Then use it here.

Reply:
https://github.com/newsboat/newsboat/pull/2561/commits/0e8adedc27881cacd0e25998cba7b91177c20775